sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -1142,6 +1199,40 @@ def __init__(self, state, provision_info):
     self.data_server.start()
     self.control_server.start()
 
+  def close(self):
+    self.control_handler.done()
+    to_wait = [
+        self.control_server.stop(self._DEFAULT_SHUTDOWN_TIMEOUT_SECS),
+        self.data_server.stop(self._DEFAULT_SHUTDOWN_TIMEOUT_SECS),
+        self.state_server.stop(self._DEFAULT_SHUTDOWN_TIMEOUT_SECS),
+        self.logging_server.stop(self._DEFAULT_SHUTDOWN_TIMEOUT_SECS)
+    ]
+    for w in to_wait:
+      w.wait()
+    GrpcServer._shared_grpc_server = None
+
+
+class GrpcWorkerHandler(WorkerHandler):
+  """An grpc based worker_handler for fn API control, state and data planes."""
+
+  def __init__(self, state, provision_info):
+    self.grpc_server = GrpcServer._shared_grpc_server
```

Comment:
The GrpcServer should be passed through WorkerHandler.create(...) to this constructor, not implicitly via a global (which may get crossed if there is more than one pipeline).